SnowplowDemo/SnowplowDemo/DemoUtils.m

```diff
@@ -139,24 +139,24 @@ + (void) trackPageViewWithTracker:(SPTracker *)tracker_ {
 + (void) trackScreenViewWithTracker:(SPTracker *)tracker_ {
     SPScreenView *event = [SPScreenView build:^(id<SPScreenViewBuilder> builder) {
         [builder setName:@"DemoScreenName"];
-        [builder setId:@"DemoScreenId"];
+        [builder setEventId:@"DemoScreenId"];
```

Comment:
I think these should be setScreenId as you are constructing a Screen View event and they need an ID as per the screen_view schema: https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.mobile/screen_view/jsonschema/1-0-0

Also, if you look in SPEvent's constructor - it sets a event id so it should already have a valid UUID set.